### PR TITLE
Version split

### LINKED ISSUE #116: Split up transpiler for different Lua Versions

### PR DESCRIPTION
**Pull request by lolleko**
Closes #116

### CODE REVIEW

File: src/Transpiler.ts [hunk 1]
```diff
@@ -18,40 +18,15 @@ export class TranspileError extends Error {
...
 export enum LuaTarget {
+    Lua51 = "5.1",
+    Lua52 = "5.2",
     Lua53 = "5.3",
     LuaJIT = "JIT",
 }
 
-export class LuaTranspiler {
+export abstract class LuaTranspiler {
     public static AvailableLuaTargets = [LuaTarget.LuaJIT, LuaTarget.Lua53];
```
Comment (apemanzilla) [inline src/Transpiler.ts]: This isn't needed any more

File: src/targets/Transpiler.52.ts [hunk 2]
```diff
@@ -0,0 +1,7 @@
+import { LuaTranspiler, TranspileError } from "../Transpiler";
+import { TSHelper as tsHelper } from "../TSHelper";
+
+import * as ts from "typescript";
+
+export class LuaTranspiler52 extends LuaTranspiler {
```
Comment (apemanzilla) [inline src/targets/Transpiler.52.ts]: Lua 5.2 added the `bit32` library, this could be used to implement bit operations

Comment (lolleko) [reply src/targets/Transpiler.52.ts]: Sure, but we should probably do that in a different PR.

File: src/Transpiler.ts (hunk 1 above)
Comment (lolleko) [reply src/Transpiler.ts]: It prevents the class from being instantiated.

File: src/targets/Transpiler.52.ts (hunk 2 above)
Comment (lolleko) [reply src/targets/Transpiler.52.ts]: Moved to #119

**Pull request merged**